extensions/namespace-lookup-management/src/main/java/io/druid/metadata/MetadataNamespaceManager.java

```diff
@@ -0,0 +1,346 @@
+    @Override
+    public void disableNamespace(final String namespace)
+    {
+      Preconditions.checkArgument(!Strings.isNullOrEmpty(namespace), "blank namespace");
+      connector.getDBI().withHandle(
+          new HandleCallback<Boolean>()
+          {
+            @Override
+            public Boolean withHandle(Handle handle) throws Exception
+            {
+              return handle.update(
+                  String.format(
+                      "UPDATE %s SET used = false WHERE namespace = :namespace",
+                      dbTables.get().getNamespacesTable()
+                  ), namespace
+              ) > 0;
+            }
+          }
+      );
+      knownNamespaces.get().remove(namespace);
+    }
+
+    @Override
+    public List<Map<String, Object>> knownNamespaces()
+    {
+      return ImmutableList.copyOf(knownNamespaces.get().values());
+    }
+
+    @Override
+    public Map<String, Object> getNamespace(String namespace)
+    {
+      Preconditions.checkArgument(!Strings.isNullOrEmpty(namespace), "blank namespace");
+      return knownNamespaces.get().get(namespace);
+    }
+
+    @Override
+    public boolean isLoadedOnce()
+    {
+      return loadedOnce.get();
+    }
+
+    @Override
+    public void registerPostPollRunnable(Runnable runnable)
+    {
+      postPollRunnable.set(Preconditions.checkNotNull(runnable, "runnable"));
+    }
+
+
+    protected void poll()
+    {
+      knownNamespaces.set(
+          connector.getDBI().withHandle(
+              new HandleCallback<ConcurrentMap<String, Map<String, Object>>>()
+              {
+                @Override
+                public ConcurrentMap<String, Map<String, Object>> withHandle(Handle handle) throws Exception
+                {
+                  final ConcurrentMap<String, Map<String, Object>> map = handle
+                      .createQuery(
+                          String.format(
+                              "SELECT payload FROM %s WHERE used ORDER BY updated_date DESC",
+                              dbTables.get().getNamespacesTable()
+                          )
+                      )
+                      .map(
+                          new ResultSetMapper<Map<String, Map<String, Object>>>()
+                          {
+                            @Override
+                            public Map<String, Map<String, Object>> map(int index, ResultSet r, StatementContext ctx)
+                                throws SQLException
+                            {
+                              try {
+                                Map<String, Object> map = jsonMapper.readValue(
+                                    r.getBytes("payload"), new TypeReference<Map<String, Object>>()
+                                    {
+                                    }
+                                );
+
+                                return ImmutableMap.<String, Map<String, Object>>of(
+                                    map.get("namespace").toString(),
```

Comment:
this is an implicit assumption that the namespaceSpec will always have key "namespace", 
add javadoc and checks in addorUpdateNamespace.